test/https.ts

```diff
@@ -20,3 +21,8 @@ test('https request with ca', withServer, async (t, server, got) => {
 	});
 	t.is(body, 'ok');
 });
+
+test('http2 works', async t => {
```

Comment:
Would be good with some more HTTP2 tests. Should be in a separate `http2.js` test file.